app/controllers/concerns/enlistment_filters.rb

```diff
@@ -76,8 +76,33 @@ def build_code_location
   def project_has_code_location?
     return unless CodeLocationSubscription.code_location_exists?(@project.id, @code_location.url,
                                                                  @code_location.branch, code_location_params[:scm_type])
-
-    flash[:notice] = t('.notice', url: @code_location.url, module_branch_name: @code_location.branch)
+    manage_deleted_enlistment
     redirect_to project_enlistments_path(@project)
   end
+
+  def manage_deleted_enlistment
+    if enlistment_deleted?
+      restore_deleted_enlistment
+    else
+      flash[:notice] = t('.notice', url: @code_location.url, module_branch_name: @code_location.branch)
+    end
+  end
+
+  def enlistment_deleted?
+    enlistment = get_enlistment
+    enlistment.try(:deleted?) ? true : false
+  end
+
+  def get_code_location_id
+    CodeLocationApi.id_from_url(@code_location.url, @code_location.branch)
+  end
+
+  def get_enlistment
+    Enlistment.where(project_id: @project.id, code_location_id: get_code_location_id).first
```

Comment:
We should use the Subscription model.

Reply:
The `code_location_id` we get it via API call from Fisbot

Reply:
The Subscription model relies upon the FDW, which will be going away entirely so that all communication is through the API. The Ohloh UI application should know about the eFISbot service (the API) and nothing about its internal workings.